### PR TITLE
playground: avoid slow parsing for Reason source ending in line comment

### PR DESCRIPTION
**Pull request by anmonteiro**
fixes https://github.com/melange-re/melange/issues/808

### CODE REVIEW

File: playground/toolchain.ml [hunk 1]
```diff
@@ -43,18 +63,12 @@ let parseWith =
...
-    let code =
-      (* Add ending new line as otherwise reason parser chokes with inputs such
-         as "//" *)
-      Js.to_string code ^ "\n"
-    in
-    try f (Lexing.from_string code)
-    with
-    (* from ocaml and reason *)
+    try f lexbuf
+    with (* from ocaml and reason *)
```
Comment (jchavarri) [inline playground/toolchain.ml:71]: Not sure if this comment makes sense anymore (at least, I don't understand it):

```suggestion
    with
```

File: playground/mel_playground.cppo.ml [hunk 2]
```diff
@@ -186,7 +186,7 @@ let compile =
     let types_signature = ref [] in
     warnings_collected := [];
     try
-      let lexbuf = Lexing.from_string str in
+      let lexbuf = Toolchain.feed_string_with_newline str in
```
Comment (jchavarri) [inline playground/mel_playground.cppo.ml:189]: Ah, so I understand this was the issue, right? We would add the newline when using the toolchain as a library, but not on the playground itself. Great finding!

**Review by jchavarri — APPROVED** [review]
❤️

File: playground/mel_playground.cppo.ml (hunk 2 above)
Comment (anmonteiro) [reply playground/mel_playground.cppo.ml:189]: yup!

**Pull request merged**